lib/floe/workflow/intrinsic_function/transformer.rb

```diff
@@ -30,9 +30,16 @@ def self.resolve_args(args)
         rule(:number   => simple(:v)) { v.match(/[eE.]/) ? Float(v) : Integer(v) }
         rule(:jsonpath => simple(:v)) { Floe::Workflow::Path.value(v.to_s, context, input) }
 
-        rule(:states_array => {:args => subtree(:args)}) { Transformer.resolve_args(args) }
+        rule(:states_array => {:args => subtree(:args)}) do
+          Transformer.resolve_args(args)
+        end
+
+        rule(:states_uuid => {:args => subtree(:args)}) do
+          args = Transformer.resolve_args(args())
+          raise ArgumentError, "wrong number of arguments to States.UUID (given #{args.size}, expected 0)" unless args.empty?
```

Comment:
I do think this should be `Floe::InvalidWorkflowError` instead of `ArgumentError`, or maybe we catch `ArgumentError` while evaluating the intrinsic_function and raise InvalidWorkflowError then?

Reply:
> I check arity (which I agree is a workflow error), then types (which feels more like a runtime error), and then specific conditions like argument must be non-zero positive (which also feels like a runtime error)

Completely agree on these